### ISSUE TITLE
Authentication Password For User

### ISSUE THREAD

**Issue by ravisoni** [opendns/autotask-php#131]
Hello,

Is there any API/Entity to check user/customer username and password is correct so we can verify our website user based on it ?

**Comment by JoshNW** [comment]
I am not sure I understand. Can you give me a bit more detail?

**Comment by KenEDirective** [comment]
Hello Josh,

I'm working with Ravisoni and what we are curious about is a way to authenticate passwords for users with portal access.

**Comment by JoshNW** [comment]
To my knowledge, there is no authentication method for ClientPortal users via the API or URL tags that would allow you to check for a valid username/password. 

These passwords are hashed in the Database and not accessible in the Autotask UI. 

I do not think that this is possible, but would be very interested to see what the other ideas are suggested on this post. Good luck!

**Comment by reynolek** [comment]
The only way i know how to valid username/password is to [fetch the zone wsdl](https://github.com/opendns/autotask-php#fetch-your-zone-wsdl) and then request Account0. It's pretty slow.

**Comment by SirPoot-** [comment]
@JoshNW is correct.

No way of accessing / verifying them. The only way to "know" is to set the contact's client portal password via the API. But then, even after you set it, you cannot verify.

Sounds like you are building out your own client portal, you will need to run your own authentication system and security mechanisms.

**Comment by SirPoot-** [comment]
@reynolek that is correct for Autotask resources (users). Their is a client portal where contacts have their own authentication. This is complete is a separate from the Autotask users.

**Comment by KenEDirective** [comment]
Thanks for the input

@SirPoot- We were not planning on building a portal however we may have too. We wanted to use what Autotask provided for a portal but we would like it so that logging into the site uses the same credentials as the ones in autotask. This would allow someone that's already logged onto the site to be able to go right into their portal without having to login a second time. I'll see what @ravisoni has to add to this and go from there.

**Comment by ravisoni** [comment]
Thanks for the input guys.
@reynolek Fetch zone info and request account not confirm us that user using right password or not in out website during login.
@JoshNW and @SirPoot What exactly we need like when user login in our website at the same time we check that user on AutoTask that user is valid and using same password which they set on AutoTask. By this way we can allow/disallow user to use our site or AutoTask section.
As per above discussion I figure we do not have such API but another side can we login in AutoTask portal by Curl so we can try to login there by Curl and check response that user is logged-in or not so we can take action on our site as per response of curl ??

**Comment by JoshNW** [comment]
To clarify, are you referring to resources in Autotask, or customers(contacts)? 

If this is to test the credentials of Autotask resources, then you could pass the username and password into an API authentication script and then react to the response. 

If you are testing customer(contact) credentials as they would appear in the client portal or taskfire, you may be out of luck using the API to check. It may be possible to work the issue from the other side, and only build client portal logins through the API. Using that method you could add users who need to access Autotask's client portal from your portal and set their password to match what ever they use to log into your portal. You may also want to set up a password reset/sync mechinism from your portal to reset their password to match your portals authentication. 

Hope that helps

**Comment by KenEDirective** [comment]
@JoshNW When you say "client portal logins through the API" Do you mean we would be able to use this username and password to actually login to the Autotask portal? I currently don't seen any way to login to the Autotask Portal unless you visit the Login page itself and entered the credentials. I may be missing something, can you expand on this a little?

Greatly appreciated.

**Comment by SirPoot-** [comment]
@KenEDirective There is no way to log in a client portal user via the API. You can only SET the client portal users credentials which is what @JoshNW  as recommending (correct me if I am wrong). So when a user logs into or creates an account on your system, you can update the client portal password to match both authentication systems creating 'synced' credentials. 

You are right, in order for them to authenticate against the the client portal they need to do they by visiting the page. There is no token or SSO solution which is why I was asking if you were building your own client portal. You pretty put run into a road block where you can only create a link to the client portal without opening up a big security issue, but at least the could be the same credentials.

**Comment by KenEDirective** [comment]
@SirPoot- Thanks for clarifying this. I was thinking this was the case but wanted to be sure I didn't misunderstand. So in the end we'd need to create our own portal or use another portal to replace the use of the Autotask one.

**Comment by SirPoot-** [comment]
@KenEDirective Depending on your needs, maybe.

@ravisoni sounds this issue has been address. Please close.
